docs/website: Improve partial evaluation / data filtering documentation (#8625)
open-policy-agent/opa: docs/docs/filtering/fragment.md

For a step-by-step walkthrough of evaluating a Rego policy _partially_, see [Evaluating a data filter policy](./partial-evaluation).
:::

> ## What is Partial Evaluation? {#what-is-partial-evaluation}

Reviewer: Unsure if this was needed? the generated fragment looks to be the same?

Author: good catch! During the process I had a link to this section at some point, which I removed again and forgot to also remove this anchor. I removed it now.

## Example Preamble

> In our running example, we'll assume a table `fruits` with columns `name`, `colour`, and `price`.

Reviewer: ```suggestion
In this running example, assume a table `fruits` exists with columns `name`, `colour`, and `price`.
```

    }
```

## Context data for Partial Evaluation

### Unknowns: database rows

Database rows are **unknown** at policy evaluation time — OPA does not have access to the database. They are represented in Rego using the convention `input.<TABLE>.<COLUMN>`, e.g. `input.fruits.name` refers to the `name` column of the `fruits` table.

The **METADATA annotation** on the policy package declares which `input` paths are unknown. OPA uses this to know which parts of the policy to leave as conditions rather than evaluate:

```rego title="policy.rego"
package filters

> # METADATA

Reviewer: This metadata comment is in a bit of a strange place? Did you mean to use the package scope?

Author: Thank you for your review @charlieegan3 ! And sorry for my late response!
I changed the metadata comments to use package scope consistently, I agree that this is more suitable / intuitive.
